sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cas-integration/cas-invoices-batches-creation.scheduler.ts

```diff
@@ -0,0 +1,54 @@
+import { InjectQueue, Processor } from "@nestjs/bull";
+import { QueueService } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import { CASInvoiceBatchService } from "../../../services";
+
+/**
+ * Scheduler to generate batches for CAS invoices for e-Cert receipts.
+ */
+@Processor(QueueNames.CASInvoicesBatchesCreation)
+export class CASInvoicesBatchesCreationScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.CASSupplierIntegration)
+    schedulerQueue: Queue<void>,
+    queueService: QueueService,
+    private readonly casInvoiceBatchService: CASInvoiceBatchService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Checks received e-Cert receipts with no invoice associated
+   * and create invoice records to be sent to CAS.
+   * @param _job process job.
+   * @param processSummary process summary for logging.
+   * @returns process summary.
+   */
+  protected async process(
+    _job: Job<void>,
+    processSummary: ProcessSummary,
+  ): Promise<string[] | string> {
+    processSummary.info("Executing CAS invoices batches creation.");
+    const createdBatch = await this.casInvoiceBatchService.createInvoiceBatch(
+      processSummary,
+    );
+    processSummary.info("CAS invoices batches creation process executed.");
+    if (!createdBatch) {
+      return "No batch was generated.";
+    }
+    return [
+      `Batch created: ${createdBatch.batchName}.`,
```

Comment:
Minor. can be called as `Batch name:`

Reply:
Unless there is a strong reason the way it is seems pretty clear to me.

Reply:
Not a blocker, but on the information we return here we say `Batch Created` where it is the batch name and `Invoices Created` where it is the invoices count.  One is name and other is count. So I just suggested. Again not a blocker and very minor.

Reply:
It says "Batch created" and uses the batch name as an identifier of what was created.
The idea is not to have property value. I will keep it as it is.